operator/fee_recipient/controller_test.go

```diff
@@ -40,109 +36,220 @@ func TestSubmitProposal(t *testing.T) {
+			share := &types.SSVShare{
+				Share: spectypes.Share{
+					ValidatorPubKey: spectypes.ValidatorPK([]byte(fmt.Sprintf("pk%046d", i))),
+					SharePubKey:     []byte(fmt.Sprintf("pk%046d", i)),
+					ValidatorIndex:  phase0.ValidatorIndex(i),
+					Committee:       []*spectypes.ShareMember{{Signer: operatorData.ID}},
+				},
+				Status:       eth2apiv1.ValidatorStateActiveOngoing,
+				OwnerAddress: ownerAddr,
+				Liquidated:   false,
+			}
+			require.NoError(t, shareStorage2.Save(nil, share))
 		}
 
-		wg.Wait()
+		frCtrl2 := NewController(logger, &ControllerOptions{
+			Ctx:               t.Context(),
+			BeaconConfig:      beaconConfig,
+			ShareStorage:      shareStorage2,
+			OperatorDataStore: operatorDataStore,
+		})
+
+		batchSize := 0
 
-		close(mockTimeChan) // Close the channel after test
-		close(mockSlotChan)
+		client := beacon.NewMockBeaconNode(ctrl)
+		client.EXPECT().SubmitProposalPreparation(gomock.Any(), gomock.Any()).DoAndReturn(
+			func(ctx context.Context, feeRecipients map[phase0.ValidatorIndex]bellatrix.ExecutionAddress) error {
+				batchSize = len(feeRecipients)
+				return nil
+			}).Times(1)
+
+		frCtrl2.beaconClient = client
+
+		err := frCtrl2.prepareAndSubmit(t.Context())
+		require.NoError(t, err)
+
+		// Should have exactly 1 batch of 500
+		require.Equal(t, 500, batchSize)
 	})
 
 	t.Run("error handling", func(t *testing.T) {
-		var wg sync.WaitGroup
-		client := beacon.NewMockBeaconNode(ctrl)
-		client.EXPECT().SubmitProposalPreparation(gomock.Any(), gomock.Any()).DoAndReturn(func(ctx context.Context, feeRecipients map[phase0.ValidatorIndex]bellatrix.ExecutionAddress) error {
-			wg.Done()
-			return errors.New("failed to submit")
-		}).MinTimes(2).MaxTimes(2)
+		// Test that errors are handled gracefully
+		db3, shareStorage3, _ := createStorageWithRecipients(t)
+		defer func() { require.NoError(t, db3.Close()) }()
+		populateStorage(t, shareStorage3, operatorData)
 
-		ticker := mocks.NewMockSlotTicker(ctrl)
-		mockTimeChan := make(chan time.Time, 1)
-		ticker.EXPECT().Next().Return(mockTimeChan).AnyTimes()
-		ticker.EXPECT().Slot().Return(phase0.Slot(100)).AnyTimes()
+		frCtrl3 := NewController(logger, &ControllerOptions{
+			Ctx:               t.Context(),
+			BeaconConfig:      beaconConfig,
+			ShareStorage:      shareStorage3,
+			OperatorDataStore: operatorDataStore,
+		})
 
-		frCtrl.beaconClient = client
-		frCtrl.slotTickerProvider = func() slotticker.SlotTicker {
-			return ticker
-		}
+		client := beacon.NewMockBeaconNode(ctrl)
+		client.EXPECT().SubmitProposalPreparation(gomock.Any(), gomock.Any()).Return(errors.New("failed to submit")).Times(1)
+
+		frCtrl3.beaconClient = client
 
-		go frCtrl.Start(t.Context())
-		mockTimeChan <- time.Now()
-		wg.Add(2)
-		wg.Wait()
-		close(mockTimeChan)
+		// Should handle error gracefully without panic
+		err := frCtrl3.prepareAndSubmit(t.Context())
+		require.NoError(t, err) // prepareAndSubmit logs errors but returns nil
```

Comment:
I think while we are touching it, we need to delete `prepareAndSubmit`'s return parameter instead of writing a comment that it always returns `nil`